forem/forem
# Update FilteredAdsQuery and user settings for opting-out of Billboards

## What type of PR is this? (check all applicable)


## Description

Per https://github.com/forem/forem-internal-eng/issues/567, we want to update the user settings to limit what kinds of ads users can opt-out of seeing. 

<img width="758" alt="221600284-11fb0349-0841-49e2-8055-2aa70d0b97e3" src="https://user-images.githubusercontent.com/2077/223448551-875e7414-6e03-4c94-bf82-4a57c5239e1f.png">

Previously, these settings applied to any/all ads, however, we want to limit these settings to apply only to `external`-type billboards. In the case of the first settings ("when browsing"), we'll hide `external`-type ads via JavaScript, meaning that users who have opted-out in this way will _sometimes_ see an `internal`- or `community`-type ad, but if the ad served up by `FilteredAdsQuery` is `external`, we'll hide it for them.

@pkfrank will provide updated copy before this can merge.

## Related Tickets & Documents

- Closes #https://github.com/forem/forem-internal-eng/issues/567

## QA Instructions, Screenshots, Recordings

* You'll want to have a local user who has articles published and at least one organization
* You'll want to have local billboards, at least one community, in-house, and external-type ad
* It can help to add `console.log` in `initializeDisplayAdVisibility` to confirm whether or not ads are being hidden
* Log-in as that user and disable one of these settings, then visit articles published by that user and articles not published by that user

### UI accessibility concerns?

Hidden ads on the client-side could conceivably still show up for folks who have JavaScript disabled, but that's likely a vanishingly small minority at this stage.

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://bookshow.blurb.com/bookshow/cache/P16417972/md/cover_2.jpeg?access_key=373fecfd3138d67e23bc458ce624e484)
